BottomSheet Info Layer added on MapFragment

oleg-popenov opened the pull request.

Commit:
BottomSheet Info Layer added on MapFragment

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/ExploreArtActivity.kt:
```diff
@@ -60,11 +84,23 @@ class ExploreArtActivity : AppCompatActivity(), ArtObjectDetailOpener {
 
         explore_floating_action_button.setOnClickListener { openSearchView() }
         search_view.findViewById(R.id.search_close_btn).setOnClickListener { closeSearchView() }
+        viewpager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener(){
```
Replace with lambda if possible

oleg-popenov replied:
There are three methods. Not possible.

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/ExploreArtActivity.kt:
```diff
@@ -60,11 +84,23 @@ class ExploreArtActivity : AppCompatActivity(), ArtObjectDetailOpener {
     override fun onStart() {
         super.onStart()
         createSearchSubscription()
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_CONTACTS)
```
We better remove this new Android 6 permissions handling for now. 
If we'll build the app with target lower than Android 6 - all permissions will be granted on the app install. Correct me if I'm wrong.

oleg-popenov replied:
As far as I understand ContextCompat will return PERMISSION_GRANTED on lower target versions. So this code won't run anyway.

genaforvena replied:
Android 6 permissions should be part of another review anyway. Remove this code.

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/ExploreArtActivity.kt:
```diff
@@ -118,6 +154,25 @@ class ExploreArtActivity : AppCompatActivity(), ArtObjectDetailOpener {
+        var artObject: ArtObjectUi = dataSource.getArtObject(id)
+        bottom_detail_author.text = artObject.authorsNames()
+        bottom_detail_name.text = artObject.name
+        if(artObject.picsUrls.isEmpty()){
```
Please use thumbPicUrl property to load the picture.

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/interfaces/InfoView.kt:
```diff
@@ -0,0 +1,7 @@
+package org.imozerov.streetartview.ui.explore.interfaces
+
+interface InfoView {
```
Rename to ArtObjectDigestViewer

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/interfaces/InfoView.kt:
```diff
@@ -0,0 +1,7 @@
+
+interface InfoView {
+
+    fun showDetails(id: String)
```
showArtObjectDigest

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/interfaces/InfoView.kt:
```diff
@@ -0,0 +1,7 @@
+interface InfoView {
+
+    fun showDetails(id: String)
+    fun hideDetails()
```
hideArtObjectDigest

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/map/ArtMapFragment.kt:
```diff
@@ -32,10 +33,23 @@ class ArtMapFragment : Fragment(), Filterable, ArtView {
         mapFragment.getMapAsync {
+            val infoView : InfoView
+            try{
+                infoView = (activity as InfoView)
```
Please cast activity in onAttach. remove field assignment in onDetach callback.

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/explore/map/ArtMapFragment.kt:
```diff
@@ -32,10 +33,23 @@ class ArtMapFragment : Fragment(), Filterable, ArtView {
+            try{
+                infoView = (activity as InfoView)
+            } catch (cce: ClassCastException) {
+                throw RuntimeException("ExploreArtActivity must implement InfoView interface")
```
Do not hardcode activity name here.
Use context.toString() here (it will be implicitly called) if you'll use $context

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/extensions/MapExtensions.kt:
```diff
@@ -28,6 +31,11 @@ fun LatLng.printableDistanceTo(point: LatLng) : String {
 }
 
 fun getCurrentLocation(context: Context): LatLng {
+    if(ContextCompat.checkSelfPermission(context,
```
Remove Android 6 permissions handling from here.

genaforvena commented on app/src/main/java/org/imozerov/streetartview/ui/extensions/MapExtensions.kt:
```diff
@@ -56,7 +64,7 @@ fun GoogleMap.addUserLocationMarker(userLocation: LatLng) {
 fun GoogleMap.addArtObject(artObject: ArtObjectUi) {
     val markerOptions = MarkerOptions().position(LatLng(artObject.lat, artObject.lng))
-            .title(artObject.authorsNames())
+            .title(artObject.id)
```
Seems weird.
I think the marker should not have title at all.

oleg-popenov replied:
As is. Marker has no fields to store our id. We need to use marker as view holder here to handle onClick properely

genaforvena replied:
So user will be able to see art object id by clicking on the marker? If so - it doesn't seem right.

Commit:
Removed setting id for title. Replaced with hashmap instead.

Commit:
Made bottom sheet to collapse on back pressed.

Fixed bug with grid view item taking all height in art object list. To fix it height attribute was moved to parent layout.

Commit:
Fixed NPE when trying to open digest for user location.

Changed art object color to green.

sanovikov71 commented on app/src/main/java/org/imozerov/streetartview/ui/explore/ExploreArtActivity.kt:
```diff
@@ -14,31 +15,46 @@ import android.support.v7.widget.SearchView
     val TAG = "ExploreArtActivity"
 
     val compositeSubscription = CompositeSubscription()
+    var bottomSheetBehavior: BottomSheetBehavior<LinearLayout>? = null
```
If I understand right, the bottom sheet logic affects map only (not gallery). So, why is it in the activity, not in the corresponding fragment?

Commit:
Added clear call to the markers map.
